cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConfiguration.java

```diff
@@ -50,4 +66,119 @@ private KafkaConfiguration(String configuration, List<String> forbiddenOptions)
     public static KafkaConfiguration unvalidated(String string) {
         return new KafkaConfiguration(string, emptyList());
     }
+
+    /**
+     * Validate the configs in this KafkaConfiguration returning a list of errors.
+     * @param kafkaVersion The broker version.
+     * @return A list of error messages.
+     */
+    public List<String> validate(KafkaVersion kafkaVersion) {
+        List<String> errors = new ArrayList<>();
+        Map<String, ConfigModel> models = readConfigModel(kafkaVersion);
+        for (Map.Entry<String, String> entry: asOrderedProperties().asMap().entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            // TODO, special parsing for "listener.security.protocol.map"
+            ConfigModel config = models.get(key);
+            if (config == null) {
+                //throw new RuntimeException("Unknown config " + key);
+            } else {
+                errors.addAll(config.validate(key, value));
+            }
+            Pattern compile = Pattern.compile("listener\\.name\\.([^\\.]+)\\.(.+)");
+            Matcher matcher = compile.matcher(key);
+            if (matcher.matches()) {
+                Set<String> listenerNames = Arrays.stream(getConfigOption("listener.security.protocol.map",
+                        "PLAINTEXT:PLAINTEXT,SSL:SSL,SASL_PLAINTEXT:SASL_PLAINTEXT,SASL_SSL:SASL_SSL").split(" *, *", -1)).map(s -> s.substring(0, s.indexOf(":")).toLowerCase(Locale.ENGLISH)).collect(Collectors.toSet());
+                String listenerName = matcher.group(1);
+                if (listenerNames.contains(listenerName)) {
+                    ConfigModel configModel = models.get(matcher.group(2));
+                    if (configModel != null) {
+                        errors.addAll(configModel.validate(key, value));
+                    }
+                } else {
+                    errors.add("No listener '" + listenerName + "' defined in 'listener.security.protocol.map'. Known listeners are " + listenerNames);
+                }
+            }
+        }
+        return errors;
+    }
+
+    private Map<String, ConfigModel> readConfigModel(KafkaVersion kafkaVersion) {
```

Comment:
Wouldn't it make sense to have this in separate class fo that we can use it also outside of Kafka configuration? I would expect that we will use it on other places to configure listeners etc.

Reply:
I would leave that until we actually have a 2nd place where we need the info. If we were going to go needing the model in multiple places we _might_ want to arrange avoiding needing to parse the JSON multiple times for each reconciliation.